pkg/didcomm/protocol/decorator/decorator.go

```diff
@@ -25,6 +25,12 @@ type Thread struct {
 	PID            string         `json:"pthid,omitempty"`
 	SenderOrder    int            `json:"sender_order,omitempty"`
 	ReceivedOrders map[string]int `json:"received_orders,omitempty"`
+	// Metadata may contain additional payload for the protocol. It might be populated by the client/protocol
+	// for outbound messages. If metadata were populated, the messenger will automatically add it to the incoming
+	// messages by the threadID. If Metadata is <nil> in the outbound message the previous payload
+	// will be added to the incoming message. Otherwise, the payload will be rewritten.
+	// NOTE: Metadata is not a part of the JSON message. The payload will not be sent to another agent.
+	Metadata map[string]interface{} `json:"metadata,omitempty"`
```

Comment:
I wonder if it would make sense to have a wrapper struct embedding an anonymous Thread struct + Metadata to more clearly delineate on-the-wire fields from fields kept within the agent.

(also if a protocol doesn't care about metadata then they can continue to use the undecorated Thread structure.)

Reply:
I am not sure, e.g when we send `msg1` we have this wrapper. When we receive a message `msg2` we cannot inject metadata. Because it does not have this wrapper structure.

Reply:
agree with @troyronda. By adding these, we are diluting the Thread struct(which is used as part of DIDComm message).

Reply:
I would place the metadata entirely outside the `~thread` decorator.

Reply:
Done!